examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,163 @@
+                    if (dataObject != null && dataObject["key"] != null)
+                    {
+                        string key = dataObject["key"].ToString();
+                        string label = dataObject["label"]?.ToString();
+                        updatedKeyLabels.Add(new KeyLabel(key, label));
+                    }
+                }
+                catch (FormatException)
+                {
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+                catch (JsonReaderException)
+                {
+                    log.LogInformation($"Queue message in invalid JSON format will be ignored.\nMessage: {decodedMessage}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static async Task<bool> BackupKeyValuesAsync(HashSet<KeyLabel> updatedKeyLabels, ILogger log)
+        {
+            bool isBackupSuccessful = false;
+            try
+            {
+                // Read all settings from primary store and update secondary store.
+                await foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettingsAsync(new SettingSelector()).ConfigureAwait(false))
+                {
+                    KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                    if (updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                    {
+                        // Current setting retrieved from primary store needs to be updated in secondary store.
+                        await secondaryAppConfigClient.SetConfigurationSettingAsync(setting).ConfigureAwait(false);
+                        log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+
+                        updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                        if (updatedKeyLabels.Count == 0)
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                // Delete key labels still present in updatedKeyLabels from secondary store.
+                foreach (var keyLabel in updatedKeyLabels)
+                {
+                    await secondaryAppConfigClient.DeleteConfigurationSettingAsync(keyLabel.Key, keyLabel.Label).ConfigureAwait(false);
+                }
+                isBackupSuccessful = true;
+            }
+            catch (RequestFailedException exception)
+            {
+                log.LogError($"Service request failed from the server.\nStatus: {exception.Status}\nError Message: {exception.Message}");
+            }
+            catch (AggregateException exception) when ((exception as AggregateException)?.InnerExceptions?.All(ex => ex is RequestFailedException) ?? false)
+            {
+                log.LogError($"Service request failed from the server with the following exceptions:");
```

Comment:
"Request to Azure App Configuration failed...."

Reply:
done